test/reverseProxy.test.js

```diff
@@ -0,0 +1,123 @@
+var constants = require('../config/constants');
+var RdGlobalConfig = constants.RdGlobalConfig;
+var nock = require('nock');
+var RdHandler = require('../src/reverseProxy');
+var http = require('http');
+var assert = require('chai').assert;
+var testHelper = require('./testHelper');
+
+
+describe('RdHandler', function () {
+  context('Reverse Proxy Server', function () {
+    var originalScheme;
+    before(function (done) {
+      this.timeout = 5000;
+      testHelper.initializeDummyLoggers();
+      testHelper.initializeDummyHandlers();
+      originalScheme = RdGlobalConfig.SCHEME;
+      RdGlobalConfig.SCHEME = 'https';
+      
+      RdHandler.startReverseProxyServer(RdGlobalConfig.RD_HANDLER_REVERSE_PROXY_PORT, function (port) {
+        console.log('Test Network Utility Reverse Proxy Server Started on Port: ', port);
+        done();
+      });
+    });
+
+    after(function (done) {
+      this.timeout = 5000;
+      RdHandler.stopReverseProxyServer(function () {
+        done();
+      });
+      testHelper.deleteLoggers();
+      testHelper.deleteHandlers();
+      nock.cleanAll();
+      RdGlobalConfig.SCHEME.restore();
+      RdGlobalConfig.SCHEME = originalScheme;
+    });
+
+    it('Requests reverse proxy on behalf of the client and returns the response', function (done) {
```

Comment:
```suggestion
    it('Requests on behalf of the client and returns the response', function (done) {
```

Reply:
Line 11:
```
context('Reverse Proxy Server', function () {
```
provides the context... so no need to mention the same in individual description.

Similarly below...

Reply:
Changed

Reply:
can be resolved..